### PR TITLE
Microplan upload

### PR DESCRIPTION
**Pull request by ashish-egov**
## Summary by CodeRabbit

- **New Features**
	- Introduced the `BulkUpload` component to enhance data upload capabilities.
	- Improved the `UploadDataCustom` component with better error handling and state management for uploaded files.

- **Bug Fixes**
	- Enhanced error reporting in the `validateData` function and improved handling of file processing errors.

- **Documentation**
	- Updated function signatures for better clarity on component functionality and error management.

### CODE REVIEW

File: health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/Module.js [hunk 1]
```diff
@@ -132,6 +133,8 @@ const componentsToRegister = {
   UpdateBoundary,
   UpdateBoundaryWrapper,
   SelectingBoundariesDuplicate,
+  XlsPreview,
+  BulkUpload,
   CampaignUpdateSummary,
   XlsPreview,
```
[the remark this answers is not in the document]
Comment (nipunarora-eGov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/Module.js:139]: @ashish-egov this is already imported

[the remark this answers is not in the document]
Comment (nipunarora-eGov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/Module.js:139]: Already registered*

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/UploadDataCustom.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
-import { Header, LoaderWithGap } from "@egovernments/digit-ui-react-components";
+import { Header, LoaderWithGap} from "@egovernments/digit-ui-react-components";
 import React, { useRef, useState, useEffect, Fragment, useMemo } from "react";
 import { useTranslation } from "react-i18next";
 import { Card, Modal, CardText } from "@egovernments/digit-ui-react-components";
-import BulkUpload from "./../../../campaign-manager/src/components/BulkUpload";
 import Ajv from "ajv";
 import XLSX from "xlsx";
 import { InfoCard, PopUp, Toast, Button, DownloadIcon, Stepper, TextBlock } from "@egovernments/digit-ui-components";
+// import XlsPreview from "../../../campaign-manager/src/components/XlsPreview";
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/UploadDataCustom.js:8]: Delete thsese comments

File: health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/Module.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashish-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/Module.js:139]: Fixed

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**